Android_Sanity.py

```diff
@@ -0,0 +1,241 @@
+import time
+import requests
+import json
+import os
+
+"""
+TODO: Wait for sample app to complete in order to actually test
+"""
+class TestCases(unittest.TestCase):
+    # vars
+    myAddress = ''
+    myBalance = ''
+    badAddress = ''
+    noTrustAddress = 'GANFGTTCZL3D477BSCPR4RMUCX6RLERFUMOKZQYWK22ZFECZ3C7WXIZK'
+    qaAccount = 'GBTVB43S7PX2EZKXTHEXAGLVTRBXDOZY2V3LOYN2PSWRIOAFPRMUO2WA'
+
+    # Set up Appium and the app
+    @classmethod
+    def setUpClass(cls):
+        # Sample App should be already installed on the emulator # TODO: add option to install
+        appPackage = 'kin.sdk.core.sample'
+        appActivity =  '.ChooseNetworkActivity'
+        cls.driver = webdriver.Remote(
+            command_executor='http://127.0.0.1:4723/wd/hub',  # Run on local server
+            desired_capabilities={
+                'appPackage': appPackage,
+                'appActivity': appActivity,
+                'platformName': 'Android',
+                'platformVersion': '7.1',
+                'deviceName': 'Nexus 5X'  # TODO: name and version from somewhere else
+            }
+        )
+        cls._values = []
+        # Timeout for element searching
+        # Official documentation says its in millisecond, but it actually works in seconds for me
+        cls.driver.implicitly_wait(10)
+
+    # Called when the test run is over
+    @classmethod
+    def tearDownClass(cls):
+        # Clear all data and close the app
+        os.system('adb shell pm clear kin.sdk.core.sample')
+
+    def findById(self,id):
+        return self.driver.find_element_by_id('kin.sdk.core.sample:id/'+id)
+
+    def findByText(self,text):
+        # yes, this is the right syntax
+        return self.driver.find_element_by_android_uiautomator('new UiSelector().text("{}")'.format(text))
+
+    # List of tests from <link to test cases>
+    def test_1_CreateAccount(self):
+        # Verify network buttons exist
+        testNetButton = self.findById('btn_test_net')
+        mainNetButton = self.findById('btn_main_net')
+
+        # Select Testnet
+        testNetButton.click()
+
+        # Verify account creation screen exists
+        createWalletButton = self.findById('btn_create_wallet')
+        createWalletButton.click()
+
+        # Account Screen
+        # TODO: Remove this after the app is fixed
+        ok = self.driver.find_element_by_id('android:id/button1')
+        ok.click()
+
+
+        # Verify that the address is fine
+        address = self.findById('public_key')
+        TestCases.myAddress = address.get_attribute('text')
+        self.assertEquals(len(TestCases.myAddress), 56)
+        self.assertEquals(TestCases.myAddress[0], 'G')
+
+    def test_2_InitialBalanceTest(self):
+        # Verify balance Labels show up
+        balanceHeader = self.findByText('getBalance:')
+        balanceLabel = self.findById('balance')
+
+        # Verify that the balance does not exist
+        time.sleep(3)  # Wait for the balance to refresh
```

Comment:
Can we get current text and wait for it to change? in case it takes more than 3 sec